### 2.1 簡單的資料型別

#### 第1題

* 整數： (c), (i), (j), (l)
* 浮點數：(d), (f), (g), (h)
* 複數：(k)
* 字串：(a), (m)
* 布林:(b), (e)

### 2.2 變數與常數

#### 第3題

In [1]:
num=12;
num=12+7.2
type(num)

float

### 2.3 常用的運算子

#### 第5題

In [2]:
2**120

1329227995784915872903807060280344576

#### 第7題

In [3]:
print(ord('y'),ord('e'))  #算出'大小寫的編碼'字元數

121 101


* 由於 kitty 和 kitten 的前四個字母都相同，所以比較第五個字元的字元碼。
* 字母 y 的字元碼為 121， e 的字元碼為 101，所以得到 False。

#### 第9題 


In [4]:
x=12; y=8   
print(bin(x)); print(bin(y))

0b1100
0b1000


In [5]:
#(a)
x&y   # and

8

In [6]:
#(b)
x|y   # or

12

In [7]:
#(c)
~x    # not

-13

In [8]:
#(d)
x^y   # xor

4

In [9]:
#(e)
x>>2  # right-shift

3

In [10]:
#(f)
x<<2  # left-shift

48

In [11]:
#(g)
~y    # not

-9

In [12]:
#(h)
y>>2

2

In [13]:
#(i)
y<<2

32

### 2.4 變數的設值與運算的優先順序

#### 第11題

In [14]:
#(a)
6+3**2*4

42

In [15]:
#(b)
15/3+4*4/2

13.0

In [16]:
#(c)
6>2+5

False

### 2.5 型別轉換

#### 第13題

In [17]:
#(a)
2+True    # 隱式，python將True轉為1，2+1=3

3

In [18]:
#(b)
6+int(17.4) # 顯式，透過int告訴Python，我們要取的是整數，因此 6+17=23

23

In [19]:
#(c)
21+6.5    # 隱式，直接把21轉成浮點數為21.0，因此 21.0+6.5=27.5

27.5

In [20]:
#(d)
str(2) + 'piggy' # 顯式，透過str告訴Python，我們要取的是字串'2'，因此得到 '2piggy'

'2piggy'

### 2.6 不同進位數字的轉換

#### 第15題
>將10進位的整數1024分別轉成2進位、8進位和16進位

In [21]:
bin(1024)   #2 進位

'0b10000000000'

In [22]:
oct(1024)   #8 進位

'0o2000'

In [23]:
hex(1024)   #16 進位

'0x400'

#### 第17題
>將2進位的數字0b1001001分別轉成8進位、10進位和16進位

In [24]:
oct(0b1001001)   # 8 進位

'0o111'

In [25]:
0b100100         # 10 進位 

36

In [26]:
hex(0b1001001)   # 16 進位

'0x49'

### 2.7 輸出與輸入函數

#### 第19題

In [27]:
a, b = 12670, 12.344

In [28]:
#(a)
print(f'a={a:>7},b={b:6.2f}')          

a=  12670,b= 12.34


In [29]:
#(b)
print(f'a={a:>+7},b={b:06.2f}')        

a= +12670,b=012.34


In [30]:
#(c)
print(f'a={a:+7,},b={b:06.1f}')        

a=+12,670,b=0012.3


In [31]:
#(d)
print(f'a={a:07,},b={b:>+6.0f}')       

a=012,670,b=   +12


In [32]:
#(e)
print(f'a={{{a:>5}}},b=\'{b:>4.0f}\'') 

a={12670},b='  12'


#### 第21題

In [33]:
num=input('input an integer:')
bin(int(num))

input an integer: 564


'0b1000110100'